### PR TITLE
Add a unique index to genres.name.

### PR DESCRIPTION
**Pull request by leonardr**
Just a quick branch to fix an old issue (https://github.com/NYPL-Simplified/server_core/issues/480).

The migration script may cause problems on servers that have multiple genres with the same name, but none of NYPL's production servers are like that, so hopefully the race condition that called attention to this problem is also gone.

### CODE REVIEW

File: model.py [hunk 1]
```diff
@@ -8708,7 +8708,7 @@ def cautious_http_get(cls, url, headers, **kwargs):
 
     # Sites that cause problems for us if we make automated
     # HTTP requests to them while trying to find free books.
-    AVOID_WHEN_CAUTIOUS_DOMAINS = ['gutenberg.org']
+    AVOID_WHEN_CAUTIOUS_DOMAINS = ['gutenberg.org', 'books.google.com']
```
Comment (leonardr) [inline model.py]: This is an unrelated change to make sure an OPDS import from unglue.it doesn't try to download PDFs from Google Books. (It's impossible for an automated client to get a PDF out of Google Books.)

**Review by courte — APPROVED** [review]
